pandas/core/computation/engines.py

```diff
@@ -93,18 +93,19 @@ def _evaluate(self):
         -----
         Must be implemented by subclasses.
         """
-        pass
+        # mypy complains if we use @abc.abstractmethod, so we do use
+        #  AbstractMethodError instead
+        from pandas.errors import AbstractMethodError
+
+        raise AbstractMethodError(self)
 
 
 class NumExprEngine(AbstractEngine):
     """NumExpr engine class"""
 
     has_neg_frac = True
 
-    def __init__(self, expr):
-        super().__init__(expr)
-
-    def convert(self):
+    def convert(self) -> str:
```

Comment:
I _think_ you could also remove this method since super().convert returns a str. could this be a remnant of py2 compat?

Reply:
I'll double-check.  I tried removing all of the `encoding` kwargs and found that io.pytables still relies on them.